ci/benchmark_bencher_baseline.yml

```
@@ -0,0 +1,68 @@
+include:
+  - local: 'ci/base.yml'
+
+.benchmark_model_stencils:
+  stage: benchmark
+  script:
+    - curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+    - source "$HOME/.cargo/env"
+    - cargo install --git https://github.com/bencherdev/bencher --branch main --locked --force bencher_cli # Install bencher CLI
+    - uv sync --no-dev --extra=dace --extra=io --extra=testing --extra=$ICON4PY_NOX_UV_CUSTOM_SESSION_EXTRAS --group=test
+    - source .venv/bin/activate
+    - BENCHER_RESULTS_JSON="results_${CI_COMMIT_SHA}_${BACKEND}_${GRID}.json"
+    - | # Run the benchmarks
+      bencher run \
+      --project "$BENCHER_PROJECT" \
+      --token "$BENCHER_API_TOKEN" \
+      --branch main \
+      --testbed "ci-runner:${SYSTEM_NAME}:${BACKEND}:${GRID}" \
+      --threshold-measure latency \
+      --threshold-test percentage \
+      --threshold-max-sample-size 64 \
+      --threshold-upper-boundary 0.1 \
+      --thresholds-reset \
+      --err \
+      --adapter python_pytest \
+      --file $BENCHER_RESULTS_JSON \
+      "pytest --benchmark-json $BENCHER_RESULTS_JSON -v --benchmark-only --backend=$BACKEND --grid=$GRID ./model"
```

I haven't measured the accuracy of the measurements over time without warm up iterations but we could also use:
```suggestion
      "pytest --benchmark-json $BENCHER_RESULTS_JSON -v --benchmark-only --backend=$BACKEND --grid=$GRID ./model --benchmark-warmup=on --benchmark-warmup-iterations=30"
```
For GPU runs specifically it might be helpful (although kernel time measurements are not precise atm)